Add knowledge base when use Traefik 2.x as ingress controller (#289)
longhorn/website: content/kb/troubleshooting-traefik-2.x-as-ingress-controller.md

---
title: "Troubleshooting: Use Traefik 2.x as ingress controller"
author: JenTing Hsiao
draft: false
date: 2021-04-20
categories:
  - "longhorn-ui"
---

## Applicable versions

All Longhorn versions with Traefik 2.x as ingress controller.

Note that, the CORS problem would not happen with Traefik 1.x as ingress controller.

## Symptoms

The Longhorn GUI frontend API requests sometimes failed to reach longhorn-manager backend.

## Reason

> The API requests would change the protocol between HTTPS/WSS, change the protocol between HTTPS/WSS would lead to a CORS problem.

Reviewer: ```suggestion
The API requests would change the protocol between HTTPS/WSS, and the change would lead to a CORS problem.
```

## Solution

> The Traefik 2.x ingress controller does not setting the websocket headers.

Reviewer: ```suggestion
The Traefik 2.x ingress controller does not set the WebSocket headers.
```

> 1. We add the following middleware to the route for the Longhorn GUI.

Reviewer: ```suggestion
1. Adds the following middleware to the route for the Longhorn frontend service.
```
